drivers/mysql/lock.go

```diff
@@ -0,0 +1,194 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-morph/morph/drivers"
+	"github.com/pkg/errors"
+)
+
+// Mutex is similar to sync.Mutex, except usable by morph to lock the db.
+//
+// Pick a unique name for each mutex your plugin requires.
+//
+// A Mutex must not be copied after first use.
```

Comment:
We can use the noCopy convention to help go vet catch this: https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/sync/cond.go;l=89;drc=refs%2Ftags%2Fgo1.17.3.

Reply:
TIL